mne/realtime/base_client.py

```diff
@@ -0,0 +1,193 @@
+# Authors: Teon Brooks <teon.brooks@gmail.com>
+#          Mainak Jas <mainak@neuro.hut.fi>
+#
+# License: BSD (3-clause)
+
+import threading
+import time
+import numpy as np
+
+from ..utils import logger, fill_doc
+
+
+def _buffer_recv_worker(client):
+    """Worker thread that constantly receives buffers."""
+    try:
+        for raw_buffer in client.iter_raw_buffers():
+            client._push_raw_buffer(raw_buffer)
+    except RuntimeError as err:
+        # something is wrong, the server stopped (or something)
+        client._recv_thread = None
+        print('Buffer receive thread stopped: %s' % err)
+
+
+class _BaseClient(object):
+    """Base Realtime Client.
+
+    Parameters
+    ----------
+    identifier : str
+        The identifier of the server. IP address or LSL id or raw filename.
+    port : int | None
+        Port to use for the connection.
+    wait_max : float
+        Maximum time (in seconds) to wait for real-time buffer to start.
+    tmin : float | None
+        Time instant to start receiving buffers. If None, start from the latest
+        samples available.
+    tmax : float
+        Time instant to stop receiving buffers.
+    buffer_size : int
+        Size of each buffer in terms of number of samples.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+    """
+
+    def __init__(self, identifier, port=None, wait_max=10., tmin=None,
```

Comment:
should this also accept an `info` argument to be consistent with the other client objects ?

Reply:
hmm...i am looking at the FieldtripClient, the MockRTClient, and StimClient, and they all have different `__init__` signatures. do you think we should standardize the pattern so they all have at least the beginning arguments? also, I feel like `info` shouldn't be the first argument in the call. i'm happy to include it but it doesn't seem that intuitive to have it there given that it will often be None and then guess. seems to be more of a secondary argument

Reply:
I mean it's an optional argument in the other clients right? It would help to have it I think. It's very useful in the case of FieldTripClient because the header strips away most of the info and the guessing doesn't always work to contruct a correct info object

Reply:
looking more into the FieldTripClient, it says that the info can be a dict but if it's none, it creates a mne.Info class. This shouldn't be allowed, right?

Reply:
Exactly, I think we should eventually deprecate the use of `None`. The guessing makes it very [hard to use](https://github.com/mne-tools/mne-python/issues/5878).

Reply:
included the info. It makes sense, especially if the stream you expect didn't include the proper metadata when you created the LSL stream, you can now correct for it with the info when launching the client